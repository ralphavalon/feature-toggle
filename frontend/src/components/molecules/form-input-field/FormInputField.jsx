import React, { useState, useEffect } from 'react';
import { Form } from 'react-bootstrap';
import PropTypes from 'prop-types';

const FormInputField = ({
  label, name, inputType, className,
  disabled, value, placeholder, inputProps
}) => {

  const [fieldValue, setFieldValue] = useState(value);
  useEffect(() => setFieldValue(value), [value]);

  const formInputProps = inputType === 'textarea' ? { as: inputType, ...inputProps } : { type: inputType, ...inputProps };

  return (
    <Form.Group className={className}>
      <Form.Label>{label}</Form.Label>
      <Form.Control placeholder={placeholder} value={fieldValue || ''} name={name} disabled={disabled}
        onChange={e => setFieldValue(e.target.value)} {...formInputProps} />
    </Form.Group>
  );
};

FormInputField.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  inputProps: PropTypes.object,
  inputType: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
};

export default FormInputField;