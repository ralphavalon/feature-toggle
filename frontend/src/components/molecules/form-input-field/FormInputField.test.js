import React from 'react';
import { shallow } from 'enzyme';

import FormInputField from './FormInputField';

afterEach(() => {
  jest.clearAllMocks();
});

const formInputFieldRequiredProps = {
  label: 'Label',
  name: 'field-name'
};

let formInputFieldOptionalProps = {
  className: 'className',
  disabled: true,
  inputProps: { rows: 3 },
  inputType: 'textarea',
  value: 'value',
  invalidMessage: 'invalidMessage'
};

describe('FormInputField', () => {
  it('renders correctly with required props', () => {
    const wrapper = shallow(<FormInputField {...formInputFieldRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders correctly with required props and optional props', () => {
    const wrapper = shallow(<FormInputField {...{...formInputFieldRequiredProps, ...formInputFieldOptionalProps}} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should change state when onChange', () => {
    const wrapper = shallow(<FormInputField {...formInputFieldRequiredProps} />);
    wrapper.find('FormControl').simulate('change', { target: { value: 'myValue' }});
    expect(wrapper.find('FormControl').prop('value')).toEqual('myValue');
    expect(wrapper).toMatchSnapshot();
  });
})

