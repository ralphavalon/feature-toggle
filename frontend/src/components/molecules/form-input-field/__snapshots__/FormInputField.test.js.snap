// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputField renders correctly with required props 1`] = `
<FormGroup>
  <FormLabel
    column={false}
    srOnly={false}
  >
    Label
  </FormLabel>
  <FormControl
    name="field-name"
    onChange={[Function]}
    value=""
  />
</FormGroup>
`;

exports[`FormInputField renders correctly with required props and optional props 1`] = `
<FormGroup
  className="className"
>
  <FormLabel
    column={false}
    srOnly={false}
  >
    Label
  </FormLabel>
  <FormControl
    as="textarea"
    disabled={true}
    name="field-name"
    onChange={[Function]}
    rows={3}
    value="value"
  />
</FormGroup>
`;

exports[`FormInputField should change state when onChange 1`] = `
<FormGroup>
  <FormLabel
    column={false}
    srOnly={false}
  >
    Label
  </FormLabel>
  <FormControl
    name="field-name"
    onChange={[Function]}
    value="myValue"
  />
</FormGroup>
`;
